### PR TITLE
noc file change for sonar security

### PR DESCRIPTION
**Pull request by OgunyemiO**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/EXUI-152

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Review by RiteshHMCTS — APPROVED** [review]

**Review by olusegz07 — APPROVED** [review]

File: src/noc/containers/noc-field/utils/noc-validators.ts [hunk 1]
```diff
@@ -1,188 +1,108 @@
...
-            return { date: true, day: true, valid: false };
-          }
-        } else {
-          return { date: true, day: true };
-        }
+      const value = control.value.toString();
+      const [year, month, day] = value.split('-').map(Number);
+      const isValidDate = moment(value, 'YYYY-MM-DD', true).isValid();
+      if (!isValidDate || year === 0 || month === 0 || month > 12 || day === 0 || day > 31) {
+        return { month: true, date: true, valid: false };
```
Comment (andywilkinshmcts) [inline src/noc/containers/noc-field/utils/noc-validators.ts:69]: The old code could return {date: true: day: true } , {date: true, year:true } or {date: true, month: true }and this code no longer does that. I think you probably need to retain this, though I'm not sure how the return value is used

File: src/noc/containers/noc-field/utils/noc-validators.ts [hunk 2]
```diff
@@ -1,188 +1,108 @@
...
-        }
-      }
-      const momentValue = control.value.toString().replace('T', ' ').replace('.000', '');
-      if (!moment(momentValue, 'YYYY-MM-DD HH:mm:ss', true).isValid()) {
-        return { datetime: true };
+      const [year, month, day] = datePart.split('-').map(Number);
+      const [hour, minute, second] = (timePart || '').split(':').map(Number);
+
+      if (!isValidDateTime || year === 0 || month === 0 || month > 12 || day === 0 || day > 31 ||
+          hour > 23 || minute > 59 || (second !== undefined && second > 59)) {
```
Comment (andywilkinshmcts) [inline src/noc/containers/noc-field/utils/noc-validators.ts:87]: Same comment as above - it doesn' return the success values any more

File: src/noc/containers/noc-field/utils/noc-validators.ts [hunk 3]
```diff
@@ -1,188 +1,108 @@
...
-            return { time: true, second: true, valid: false };
-          }
-        } else {
-          return { time: true, second: true };
-        }
+      const value = control.value.toString();
+      const [hour, minute, second] = value.split(':').map(Number);
+      const isValidTime = moment(value, 'HH:mm:ss', true).isValid();
+      if (!isValidTime || hour > 23 || minute > 59 || (second !== undefined && second > 59)) {
+        return { hour: true, time: true, valid: false, message: '' };
```
Comment (andywilkinshmcts) [inline src/noc/containers/noc-field/utils/noc-validators.ts:103]: And again

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
Your refactoring isn't functiionally equivalent, so unless you are confident that this doesn't matter (and have tests to prove it) it's probably not correct

File: src/noc/containers/noc-field/utils/noc-validators.ts (hunk 1 above)
Comment (OgunyemiO) [reply src/noc/containers/noc-field/utils/noc-validators.ts:69]: @andywilkinshmcts, the above code (date validations) works as intended when I ran some test on it locally. Please see the attached date validations images attached.

Screenshot 2024-07-26 at 15 40 17
Screenshot 2024-07-26 at 15 41 00
Screenshot 2024-07-26 at 15 41 24
Screenshot 2024-07-26 at 15 41 54

File: src/noc/containers/noc-field/utils/noc-validators.ts (hunk 2 above)
Comment (OgunyemiO) [reply src/noc/containers/noc-field/utils/noc-validators.ts:87]: Please see my comments above. It's working as intended.

File: src/noc/containers/noc-field/utils/noc-validators.ts (hunk 3 above)
Comment (OgunyemiO) [reply src/noc/containers/noc-field/utils/noc-validators.ts:103]: Please see my comments above. It's working as intended.

**Review by andywilkinshmcts — DISMISSED** [review]

**Pull request closed without merging**